apps/api/src/app/workflows-v2/usecases/upsert-workflow/validate-and-persist-workflow-issues.usecase.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  ContentIssue,
+  RuntimeIssue,
+  StepIssueEnum,
+  StepIssuesDto,
+  WorkflowIssueTypeEnum,
+  WorkflowResponseDto,
+} from '@novu/shared';
+import {
+  ControlValuesEntity,
+  NotificationStepEntity,
+  NotificationTemplateEntity,
+  NotificationTemplateRepository,
+} from '@novu/dal';
+import { Injectable } from '@nestjs/common';
+import { ValidateWorkflowCommand } from './validate-workflow.command';
+import { ValidateControlValuesAndAddDefaultsUseCase } from '../validate-control-values/validate-control-values-and-add-defaults.usecase';
+import { WorkflowNotFoundException } from '../../exceptions/workflow-not-found-exception';
+
+@Injectable()
+export class ValidateAndPersistWorkflowIssuesUsecase {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private validateControlValuesAndAddDefaultsUseCase: ValidateControlValuesAndAddDefaultsUseCase
+  ) {}
+
+  async execute(command: ValidateWorkflowCommand): Promise<NotificationTemplateEntity> {
+    const workflowIssues = await this.validateWorkflowBody(command);
+    const stepIssues = this.validateSteps(command.workflow.steps, command.stepIdToControlValuesMap);
+    const workflowWithIssues = this.updateIssuesOnWorkflow(command.workflow, workflowIssues, stepIssues);
+    await this.persistWorkflow(command, workflowWithIssues);
+    console.log('workflowWithIssues', workflowWithIssues.issues);
+
+    const workflow = await this.getWorkflow(command);
+    console.log('workflowWithIssues2', workflow.issues);
+
+    return workflow;
+  }
+
+  private async persistWorkflow(command: ValidateWorkflowCommand, workflowWithIssues) {
+    await this.notificationTemplateRepository.update(
+      {
+        _id: command.workflow._id,
+        _environmentId: command.user.environmentId,
+      },
+      {
+        ...workflowWithIssues,
+      }
+    );
+  }
+  private async getWorkflow(command: ValidateWorkflowCommand) {
+    const newVar = await this.notificationTemplateRepository.findById(command.workflow._id, command.user.environmentId);
+    if (newVar == null) {
```

Comment:
The error checking should be required at this point.

Reply:
it has nothing to do with error checking, that's the contract with the repo, i'm renaming and validating with a clear error